File: task_equivalents/llm_translations/gemini/optimized/python/data_visualization/01_scatter_trend_grouped.py
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

auto = pd.read_csv('data/csv/Auto.csv', na_values='?')
auto = auto.dropna()

X = auto['horsepower'].values
Y = auto['mpg'].values

b, a = np.polyfit(X, Y, 1)
yfit = a + b * X

plt.scatter(X, Y)
plt.plot(X, yfit, color='red')
plt.xlabel('horsepower')
plt.ylabel('mpg')
plt.show()

File: task_equivalents/llm_translations/gemini/optimized/r/data_visualization/01_scatter_trend_grouped.R
Auto <- read.csv('data/csv/Auto.csv', na.strings = '?')
Auto <- na.omit(Auto)

model1_fit <- lm(mpg ~ horsepower, data = Auto)

plot(Auto$horsepower, Auto$mpg, xlab = 'horsepower', ylab = 'mpg')
abline(model1_fit, col = 'red')